libtxref/txref.cpp

```diff
@@ -384,6 +390,7 @@ namespace txref {
     DecodedResult decode(const std::string & txref) {
 
         std::string txrefClean = bech32::stripUnknownChars(txref);
+        txrefClean = convertToLowercase(txrefClean);
```

Comment:
Is the old friendly "be liberal in what you accept" appropriate here?  I think we should technically fail, but spew a helpful error message that the user is forced to parse for themselves.  They are capable of lowercasing the string if they want to be liberal.

Reply:
Yeah we could go either way. The bech32 spec (bip173) specifies "Decoders MUST NOT accept strings where some characters are uppercase and some are lowercase" so we know that will fail. However, being liberal here is pretty simple since this is an "easy fix". Let's discuss further.

Reply:
The new standard is, "Be liberal in what your error console explains, but do not accept off-spec data."